### DISCUSSION TITLE
On opening data tab of table, set focus on filter

### DISCUSSION THREAD

**Discussion by Khartir** [dbeaver/dbeaver#23401] [Ideas]
**Is your feature request related to a problem? Please describe.**
When opening the data tab of a table, the usual thing I want to do when I type something, is to filter the table. However, the initial focus of the tab is on the first field in the table. So if I type something, instead of filtering I edit the existing data. I have done this numerous times, when rapidly switching between tables while searching for something.
IMHO that is not useful in any way. If I want to change some data, the chances that it is exactly the first field are almost 0, so I would have to change the field anyway.

**Describe the solution you'd like**
When opening the data tab of a table (or switching to it from another table), set the focus/cursor in the filter input, rather than the first field of actual data.

**Comment by E1izabeth** [comment]
Thank you for sharing your use case. However, I think, it's controversial. Some users don't use this filter text field at all. Some open the data viewer to observe data only, and build filters automatically with UI.

**Comment by Khartir** [comment]
> Some open the data viewer to observe data only, and build filters automatically with UI.

Sure, but they wouldn't care either way, would they?

**Comment by E1izabeth** [comment]
I think, that having focus on filter text field first is a usability issue for blind people

**Comment by arhayka** [comment]
Hello @Khartir 
You can try to use the shortcut to open the filter: `Ctrl+Alt+Shift+T ` for Windows, `⌃⌥⇧T `for macOS.
You can change it to something more useful in Preferences —> Keys:

<img width="940" alt="Screenshot 2024-04-11 at 14 56 40" src="https://github.com/dbeaver/dbeaver/assets/12581569/f9689b91-6a09-4eeb-a974-78a7829c2069">
